syntax = "proto3";

package messages;

import "proto/structs.proto";

// Transaction
message TransactionRequest {
    uint32 clientId = 1;
    uint64 messageId = 2;
}

message TransactionReply {
    ReplyCode code = 1;
}

// Request for the datacenter name
message DatacenterRequest {}

message DatacenterReply {
    string datacenter = 1;
}

// Request for the message log of the server
message LogRequest {}

message LogReply {
    ReplyCode code = 1;
    string address = 2;
    repeated string log = 3;
}

service Client {
    rpc execute(TransactionRequest) returns (TransactionReply) {}
    rpc datacenter(DatacenterRequest) returns (DatacenterReply) {}
    rpc log(LogRequest) returns (LogReply) {}
}